term-entry: kotlin filter() (#3773)
Codecademy/docs: content/kotlin/concepts/strings/terms/filter/filter.md

---

> Title: 'filter()'

Reviewer: ```suggestion
Title: '.filter()'
```

> Description: 'filter elements from collection based on a given predicate.'

Reviewer: ```suggestion
Description: 'Filters elements from collection based on a given predicate.'
```

Subjects:
  - 'Computer Science'

>   - 'Web Development'

Reviewer: ```suggestion
  - 'Code Foundations'
```
Web dev doesn't make sense here.

Tags:
  - 'Collections'
  - 'Filter'
  - 'Strings'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

> 
> ## Syntax
> 

Reviewer: ```suggestion

```
this should not come here.

> The **filter()** function is used to filter elements from a collection based on a given _predicate_. It returns a new collection containing only elements which statifies a specified condition.

Reviewer: ```suggestion
The **`filter()`** function is used to filter elements from a collection based on a given `predicate`. It returns a new collection containing only elements which satisfies a specified condition.
```

> 

Reviewer: ```suggestion
## Syntax 

```

```pseudo
str.filter(predicate: (char) -> Boolean)

> 

Reviewer: ```suggestion
```

```

> Given a string `str` above, `filter()` method can filter the characters from the string using a predicate which is passed as an argument.

Reviewer: ```suggestion
- When working with the string `str` mentioned above, the `filter()` method allows selective removal of characters from the string based on a provided predicate.
```

Author: @dakshdeepHERE, `filter()` method does not remove the elements from the string instead it creates a new collection which includes only selective elements

Reviewer: Yes let me refactor it. It basically create a new collection that includes only the elements that satisfy a specified predicate.

Author: yes, I have made the changes as you suggested also this one. You can check it on my latest commit.

> 
> ### Predicate
> 
> Predicates are lambda functions that takes an element of the collection as an argument and returns a `Boolean` value. The filter() function will include the elements for which the predicate returns `true` in the resulting collection.

Reviewer: ```suggestion

- `Predicates` are lambda functions that takes an element of the collection as an argument and returns a `Boolean` value. The **`filter()`** function will include the elements for which the predicate returns `true` in the resulting collection.
```
No need to create a heading for this.

## Example

In this example, the goal is to filter only letters from the string `str` by using the `string.filter()` method.

```kotlin
fun main(args: Array<String>) {
    val str = "Hello, World!"
    val result = str.filter({ it -> it.isLetter() })
    println("Filtered String : " + result)
}
```

> Output of the above code will be -

Reviewer: ```suggestion
Output of the above code will be:
```

> ```pseudo

Reviewer: ```suggestion
```shell
```
use the tag shell for code output
